### PR TITLE
Ft login user via email and password 182201718

### PR DESCRIPTION
**Pull request by alaintresor**
#### What does this PR do?
- add user login functionality via email and password
- add swagger documentation for this endpoint

#### Description of Task to be completed?
-user sign up functionality

#### How should this be manually tested?
after clone, it, cd into it and run npm install, npm run start:dev and open any web browser type in this UR "localhost:['YOUR_PORT']/api/v1/docs/" and then you found user route

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/182201718

Screenshot (30)
Screenshot (31)

### CODE REVIEW

**Review by Placide123 — DISMISSED** [review]

**Comment by Patricknshimiyimana** [comment]
This branch has conflicts that must be resolved

File: src/services/user.service.js [hunk 1]
```diff
@@ -37,3 +38,35 @@ export const createUser = async (req, res) => {
 	}
 };
 
+
+export const loginUser= async (req, res) => {
```
Comment (joshwambere) [inline src/services/user.service.js]: Please move this codes into Authentication controller as registration is in authentication controller

File: src/services/user.service.js [hunk 2]
```diff
@@ -37,3 +38,35 @@ export const createUser = async (req, res) => {
...
+		const user = await users.findOne({
+			where:{
+				email,
+				password
+			}
+		,attributes: {exclude: ['password','createdAt','updatedAt']}});
+		if(!user){
+			throw new Error('User not found');
+		}
+		return res.status(200).json({
```
Comment (joshwambere) [inline src/services/user.service.js]: where do you compare the provided password and the actual password in DB

**Review by joshwambere — CHANGES_REQUESTED** [review]
and also write tests for you endpoint

**Comment by alaintresor** [comment]
Screenshot (32)

File: .env.example [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
-#Postgres setup
-PGPORT=YOUR_POSTGRES_PORT #ex: 5432
-PGPASSWORD=YOUR_POSTGRES_PASSWORD
-PGUSER=YOUR_POSTGRES_USERNAME
-PGHOST=YOUR_POSTGRES_HOSTNAME #pc_ex= localhost | #docker_ex= node_db 
-
-#Postgres environment database name
-DEV_PGDATABASE=YOUR_POSTGRES_DEV_DATABASE
-TEST_PGDATABASE=YOUR_POSTGRES_TEST_DATABASE
-PROD_PGDATABASE=YOUR_POSTGRES_PROD_DATABASE
```
Comment (uwambajeddy) [inline .env.example]: It wasn't necessary to  change this structure

File: README.md [hunk 4]
```diff
@@ -1,3 +1 @@
-[![Coverage Status](https://coveralls.io/repos/github/atlp-rwanda/bravo-bn-be/badge.svg?branch=develop)](https://coveralls.io/github/atlp-rwanda/bravo-bn-be?branch=develop)
-
-# bravo-bn-be
+# bravo-bn-be
```
Comment (uwambajeddy) [inline README.md]: Bro you removed the coverage badge

File: package.json [hunk 5]
```diff
@@ -12,8 +12,7 @@
     "seed": "npx sequelize-cli db:seed:all",
     "undo-seed": "npx sequelize-cli db:seed:undo",
     "test": "cross-env NODE_ENV=test npm run down && npm run migrate && npm run seed && mocha --require @babel/register --timeout 30000 --exit",
-    "coverage": "nyc npm run test | coveralls",
-    "prepare": "husky install"
```
Comment (uwambajeddy) [inline package.json]: you removed this command

File: src/controllers/error.js [hunk 6]
```diff
@@ -1,4 +1,7 @@
-import AppError from '../utils/appError';
```
Comment (uwambajeddy) [inline src/controllers/error.js:1]: leave this code as it is.

File: src/database/models/user.js [hunk 7]
```diff
@@ -21,14 +21,19 @@ export default (sequelize, DataTypes) => {
...
     email: DataTypes.STRING,
     password: DataTypes.STRING,
-    birthDate: DataTypes.DATE,
     phoneNumber: DataTypes.STRING,
+    image: DataTypes.STRING,
     passwordChangedAt: DataTypes.DATE,
     passwordResetExpires: DataTypes.DATE,
-    phoneNumber: DataTypes.STRING,
-    passwordChangedAt: DataTypes.DATE,
-    passwordResetExpires: DataTypes.DATE,
```
Comment (uwambajeddy) [inline src/database/models/user.js]: but whyyyy?

File: src/migrations/20220523191126-create-users.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
+'use strict';
```
Comment (joshwambere) [inline src/migrations/20220523191126-create-users.js]: migrations folder should be in Database folder

File: src/services/user.service.js [hunk 9]
```diff
@@ -3,18 +3,19 @@ const users = db['users']
 
 export const getAllUsers = async (req, res) => {
  
-	try{
-			users.findAndCountAll().then(users => {
+	// try{
```
Comment (joshwambere) [inline src/services/user.service.js:7]: same with seeders

File: src/controllers/error.js [hunk 10]
```diff
@@ -1,4 +1,7 @@
...
+
+import { defaults } from 'joi';
+import AppError from '../utils/AppError';
```
Comment (joshwambere) [inline src/controllers/error.js]: imports error here

**Review by joshwambere — CHANGES_REQUESTED** [review]

**Review by Placide123 — DISMISSED** [review]

**Review by Patricknshimiyimana — APPROVED** [review]

**Review by Placide123 — APPROVED** [review]

**Review by uwambajeddy — APPROVED** [review]

**Pull request merged**